lib/jwe/encrypt.js

```diff
@@ -369,15 +369,14 @@ function JWEEncrypter(cfg, fields, recipients) {
           return jwe;
         } else if (props.zip === "DEF") {
           return new Promise(function(resolve, reject) {
-            zlib.deflateRaw(Buffer.from(pdata, "binary"), function(err, data) {
-              if (err) {
-                reject(err);
-              }
-              else {
-                jwe.plaintext = data;
-                resolve(jwe);
-              }
-            });
+            try {
+              var data = pako.deflateRaw(Buffer.from(pdata, "binary"));
+
+              jwe.plaintext = data;
```

Comment:
this `data` ought to be wrapped with `Buffer.from` to keep the output as a node.js Buffer.